edu-services/gradebook-service/impl/src/java/org/sakaiproject/component/gradebook/GradebookServiceHibernateImpl.java

```diff
@@ -3113,6 +3117,19 @@ public void updateGradebookSettings(String gradebookUid, GradebookInformation gb
 		//If category does not have an ID it is new
 		//If category has an ID it is to be updated. Update and remove from currentCategoryMap.
 		//Any categories remaining in currentCategoryMap are to be removed.
+		Collections.sort(newCategoryDefinitions, new Comparator<CategoryDefinition>() {
```

Comment:
Could you stick an additional doc line in here just saying that we need to sort them first so that new ones get created in the correct order?